tez-runtime-library/src/main/java/org/apache/tez/runtime/library/common/CompositeInputAttemptIdentifier.java

```diff
@@ -50,6 +50,12 @@ public InputAttemptIdentifier expand(int inputIdentifierOffset) {
     return new InputAttemptIdentifier(getInputIdentifier() + inputIdentifierOffset, getAttemptNumber(), getPathComponent(), isShared(), getFetchTypeInfo(), getSpillEventId());
   }
 
+  public boolean include(int thatInputIdentifier, int thatAttemptNumber) {
+    return
+        super.getInputIdentifier() <= thatInputIdentifier && thatInputIdentifier < (super.getInputIdentifier() + inputIdentifierCount) &&
```

Comment:
can you reuse e.g. guava's Range for better readability?
https://www.geeksforgeeks.org/range-class-guava-java/

Reply:
Sure, we can. Modified to use guava's Range.